### PR TITLE
feat: Added a Glue job example for a Step Functions workflow

### PR DESCRIPTION
**Pull request by heeki**
*Description of changes:*
Template for creating a Step Functions workflow that executes a Glue Job, waits for completion, and then invokes a Lambda function on success or failure.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python3.9/cookiecutter-aws-sam-step-functions-etl-python/{{cookiecutter.project_name}}/functions/etl_check/app.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+client = session.client("glue")
+
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, datetime):
+            return o.isoformat()
+        return json.JSONEncoder.default(self, o)
+
+def lambda_handler(event, context):
+    """Sample pure Lambda function
```
Comment (c2tarun) [inline python3.9/cookiecutter-aws-sam-step-functions-etl-python/{{cookiecutter.project_name}}/functions/etl_check/app.py:15]: Should this comment be updated to explain what this method/lambda function is doing?

File: python3.9/cookiecutter-aws-sam-step-functions-etl-python/{{cookiecutter.project_name}}/functions/etl_failure/app.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+def lambda_handler(event, context):
+    """Sample pure Lambda function
```
Comment (c2tarun) [inline python3.9/cookiecutter-aws-sam-step-functions-etl-python/{{cookiecutter.project_name}}/functions/etl_failure/app.py:2]: Same here.

File: python3.9/cookiecutter-aws-sam-step-functions-etl-python/{{cookiecutter.project_name}}/functions/etl_success/app.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+def lambda_handler(event, context):
+    """Sample pure Lambda function
```
Comment (c2tarun) [inline python3.9/cookiecutter-aws-sam-step-functions-etl-python/{{cookiecutter.project_name}}/functions/etl_success/app.py:2]: Same here.

**Review by c2tarun — APPROVED** [review]
Please update docstring of individual lambda functions. Apart from that it LGTM!!.

**Review by qingchm — APPROVED** [review]
LGTM

**Pull request merged**